[llvm] build without llvm option fails

**Building option -DTI_WITH_LLVM:BOOL=OFF is broken**
Due to some recent changes, this option raises build errors. At this moment, ndarray and JITSession are using "LlvmProgramImpl", which will raise not defined error in the latest build system.

> @PGZXB [7bae9c7](https://github.com/taichi-dev/taichi/commit/7bae9c77f69e4215e809ee6ddcfeb9f45c724842) <- this works for me with LLVM OFF.

Are you sure you built successfully with `-DTI_WITH_LLVM:BOOL=OFF`?
I try on https://github.com/taichi-dev/taichi/commit/7bae9c77f69e4215e809ee6ddcfeb9f45c724842. But I get error : 
```shell
ndarray.h:57:3: error: unknown type name 'LlvmProgramImpl'; did you mean 'ProgramImpl'?
  LlvmProgramImpl *prog_impl_{nullptr};
  ^~~~~~~~~~~~~~~
  ProgramImpl
```

--- a/taichi/jit/jit_session.cpp
+++ b/taichi/jit/jit_session.cpp
@@ -35,10 +35,4 @@ std::unique_ptr<JITSession> JITSession::create(LlvmProgramImpl *llvm_prog,
 #endif
 }
 
-#ifdef TI_WITH_LLVM
-llvm::DataLayout JITSession::get_data_layout() {
-  TI_NOT_IMPLEMENTED
-}
-#endif
-
 TLANG_NAMESPACE_END

--- a/taichi/jit/jit_session.h
+++ b/taichi/jit/jit_session.h
@@ -32,7 +32,7 @@ class JITSession {
     TI_NOT_IMPLEMENTED
   }
 
-  virtual llvm::DataLayout get_data_layout();
+  virtual llvm::DataLayout get_data_layout() = 0;
 
   static std::unique_ptr<JITSession> create(LlvmProgramImpl *llvm_prog,
                                             Arch arch);

--- a/taichi/program/ndarray.h
+++ b/taichi/program/ndarray.h
@@ -16,6 +16,7 @@ namespace taichi {
 namespace lang {
 
 class Program;
+class LlvmProgramImpl;
 
 class Ndarray {
  public:

